[git-webkit] Check expiration in token (Follow-up)

JonWBedard opened the pull request:
#### 852c6efbe0594eecccd53dabf48395441b894f39
<pre>
[git-webkit] Check expiration in token (Follow-up)
<a href="https://bugs.webkit.org/show_bug.cgi?id=240883">https://bugs.webkit.org/show_bug.cgi?id=240883</a>
&lt;rdar://93859778&gt;

Reviewed by NOBODY (OOPS!).

* Tools/Scripts/libraries/webkitbugspy/webkitbugspy/github.py:
(Tracker.credentials.validater): Treat token as expired if it will expire in the next 12 hours.
</pre>

aj062 commented on Tools/Scripts/libraries/webkitbugspy/webkitbugspy/github.py:
```diff
@@ -111,7 +111,8 @@ def validater(username, access_token):
                 expiration = int(calendar.timegm(datetime.strptime(expiration, '%Y-%m-%d %H:%M:%S UTC').timetuple()))
-            if (expiration is None or expiration > time.time()) and response.status_code == 200 and response.json().get('login') == username:
+                expiring_soon = 12 * 60 * 60  # Consider a token expiring in the next 12 hours to be expired
+            if (expiration is None or expiring_soon > int(time.time())) and response.status_code == 200 and response.json().get('login') == username:
```
expiring_soon is not a delta, so `expiring_soon > int(time.time())` is always going to be False.

aj062 commented on Tools/Scripts/libraries/webkitbugspy/webkitbugspy/github.py:
```diff
@@ -111,7 +111,8 @@ def validater(username, access_token):
             if expiration:
                 expiration = int(calendar.timegm(datetime.strptime(expiration, '%Y-%m-%d %H:%M:%S UTC').timetuple()))
-            if (expiration is None or expiration > time.time()) and response.status_code == 200 and response.json().get('login') == username:
+                expiring_soon = 12 * 60 * 60  # Consider a token expiring in the next 12 hours to be expired
```
> Consider a token expiring in the next 12 hours to be expired

Why? If it's not expired yet, what's the problem in using that unexpired token?

Instead can we print a warning that token is going to expire in next few days? (maybe start 5 days before)

JonWBedard replied:
`git-webkit setup` is the command that actually runs this check. We don't do a validity check for every request to GitHub because doing a validity check involve making a request. That means we're only really running this check when folks are setting up their repository, and it seems bad to allow users to set up a repository that will be potentially invalid later that same day.

JonWBedard committed:
[git-webkit] Check expiration in token (Follow-up)
https://bugs.webkit.org/show_bug.cgi?id=240883
<rdar://93859778>

Reviewed by NOBODY (OOPS!).

* Tools/Scripts/libraries/webkitbugspy/webkitbugspy/github.py:
(Tracker.credentials.validater): Treat token as expired if it will expire in the next 12 hours.
